UpstreamDownstreamDialog.py

```diff
@@ -61,6 +63,48 @@ def __init__(self, iface, recLayer, parent=None):
         self.pushButtonSearchUpstream.clicked.connect(
             lambda: self.__addLayer(UPSTREAM_URL, "Upstream"))
 
+
+        self.pushButtonSetup.clicked.connect(self.__showSetupDialog)
+        if not(len(QgsMapLayerRegistry.instance().mapLayersByName("NIWA REC v2.0")) != 0):
```

Comment:
This breaks if the user renames the layer. Might be better to look at the source of the layer (i.e. `QgsMapLayer::source`) and test against the URL of the service.